### PR TITLE
Support designs referenced as a submodule

### LINKED ISSUE #106: Does not include designs from submodules
When setting a theme that is included as a submodule in the original Adminer repo (https://github.com/vrana/adminer/tree/master/designs) through `ADMINER_DESIGN`, Adminer falls back to the original theme. https://github.com/Niyko/Hydra-Dark-Theme-for-Adminer/issues/13 lists a few possible solutions but I think this Dockerfile should include all themes listed in the original repo.

### PR DESCRIPTION
**Pull request by TimWolla**
Resolves #106

### CODE REVIEW

File: 4/Dockerfile [hunk 1]
```diff
@@ -42,15 +42,17 @@ COPY	*.php /var/www/html/
 
 ENV	ADMINER_VERSION 4.8.1
 ENV	ADMINER_DOWNLOAD_SHA256 2fd7e6d8f987b243ab1839249551f62adce19704c47d3d0c8dd9e57ea5b9c6b3
-ENV	ADMINER_SRC_DOWNLOAD_SHA256 ef832414296d11eed33e9d85fff3fb316c63f13f05fceb4a961cbe4cb2ae8712
+ENV	ADMINER_COMMIT 1f173e18bdf0be29182e0d67989df56eadea4754
 
 RUN	set -x \
-&&	curl -fsSL https://github.com/vrana/adminer/releases/download/v$ADMINER_VERSION/adminer-$ADMINER_VERSION.php -o adminer.php \
+&&	apk add --no-cache --virtual .build-deps git \
+&&	curl -fsSL "https://github.com/vrana/adminer/releases/download/v$ADMINER_VERSION/adminer-$ADMINER_VERSION.php" -o adminer.php \
```
Comment (Reintjuu) [inline 4/Dockerfile:49]: Can't you just run https://github.com/vrana/adminer/blob/master/compile.php after cloning the git repo including the submodules? You won't have to download this release and can just compile it from the already locally existing sources.

Comment (TimWolla) [reply 4/Dockerfile:49]: I probably can, but I preferred to use the “official build” for the actual Adminer “binary”.

**Comment by Temzzzz1** [comment]
Hi, what's about this PL?

**Comment by TimWolla** [comment]
Hey @tianon. Can you PTAL at this PR? Do you consider my solution with checking out a specific commit to be acceptable with regard to security/reproducibility?

File: 4/Dockerfile [hunk 2]
```diff
@@ -42,15 +42,17 @@ COPY	*.php /var/www/html/
...
 &&	echo "$ADMINER_DOWNLOAD_SHA256  adminer.php" |sha256sum -c - \
-&&	curl -fsSL https://github.com/vrana/adminer/archive/v$ADMINER_VERSION.tar.gz -o source.tar.gz \
-&&	echo "$ADMINER_SRC_DOWNLOAD_SHA256  source.tar.gz" |sha256sum -c - \
-&&	tar xzf source.tar.gz --strip-components=1 "adminer-$ADMINER_VERSION/designs/" "adminer-$ADMINER_VERSION/plugins/" \
-&&	rm source.tar.gz
+&&	git clone --recurse-submodules=designs --depth 1 --shallow-submodules --branch "v$ADMINER_VERSION" https://github.com/vrana/adminer.git /tmp/adminer \
+&&	(cd /tmp/adminer/ && git checkout "$ADMINER_COMMIT") \
```
Comment (tianon) [inline 4/Dockerfile:52]: It throws a big warning, but I'd suggest putting `$ADMINER_COMMIT` directly in `--branch` instead (so it doesn't check out the tag and then potentially check out something else, which I don't think will refresh the submodules):

```suggestion
&&	git clone --recurse-submodules=designs --depth 1 --shallow-submodules --branch "$ADMINER_COMMIT" https://github.com/vrana/adminer.git /tmp/adminer \
```

File: 4/Dockerfile [hunk 3]
```diff
@@ -42,15 +42,17 @@ COPY	*.php /var/www/html/
...
+&&	cp -r /tmp/adminer/designs/ /tmp/adminer/plugins/ . \
+&&	rm -rf /tmp/adminer/ \
+&&	apk del .build-deps
```
Comment (tianon) [inline 4/Dockerfile:55]: I'd include `--no-network` on this `apk del`, but that's a minor preference (and shouldn't make any functional difference).

**Review by tianon — COMMENTED** [review]
General approach seems OK to me -- using the full commit hash should give pretty reasonable assurance (as good as we're going to get).

File: 4/Dockerfile (hunk 2 above)
Comment (TimWolla) [reply 4/Dockerfile:52]: I think I attempted that, but it somehow didn't work. I'll recheck this, thanks.

Comment (tianon) [reply 4/Dockerfile:52]: Now that I'm testing again, it doesn't work, hmm.  I know I've done this before.

Comment (tianon) [reply 4/Dockerfile:52]: I think to implement this fully will probably require something fiddly like `git init` + `git fetch --shallow` + `git submodule --init --update designs` :grimacing:

Comment (tianon) [reply 4/Dockerfile:52]: At that point, maybe it's worth just doing `--branch` with the tag as you do here, but then do something like `git -C /tmp/adminer rev-parse HEAD` and just verifying that it matches the commit we expected?

Comment (TimWolla) [reply 4/Dockerfile:52]: Ah, good suggestion. That's indeed cleaner than my checkout to perform the "verification".

**Pull request merged**